src/tribler-core/tribler_core/components/restapi.py

```diff
@@ -1,9 +1,59 @@
+from abc import ABC
+from typing import Optional, Set, Tuple
+
 from tribler_common.simpledefs import STATE_START_API
 from tribler_core.components.base import Component
 from tribler_core.components.reporter import ReporterComponent
 from tribler_core.exception_handler import CoreExceptionHandler
 from tribler_core.restapi.rest_manager import ApiKeyMiddleware, RESTManager, error_middleware
 from tribler_core.restapi.root_endpoint import RootEndpoint
+from tribler_core.restapi.state_endpoint import StateEndpoint
+
+
+class RestfulComponent(Component, ABC):
+    _rest_manager: Optional[RESTManager] = None
+    endpoint_attrs: Set[Tuple[str, str]]
+
+    def __init__(self):
+        super().__init__()
+        self.endpoint_attrs = set()
+
+    async def set_readable_status(self, readable_status):
+        rest_component = await self.get_component(RESTComponent)
+        if rest_component:
+            state_endpoint: StateEndpoint = rest_component.rest_manager.get_endpoint('state')
+            state_endpoint.readable_status = readable_status
+
+    async def init_endpoints(self, endpoints, values, ipv8=None, ipv8_endpoints=()):
```

Comment:
When reading through this code I thought at first that the method is modal: if `ipv8` arg is not None, act on IPv8 endpoints.

I'd rather split this method in two:
`init_endpoints` and `init_ipv8_endpoints`
Otherwise, the definition is misleading and unobvious

Reply:
Agree 👍

Reply:
Done